code/game/machinery/doppler_array.dm

```diff
@@ -111,22 +123,74 @@ GLOBAL_LIST_EMPTY(doppler_arrays)
 		return FALSE
 	if(!istype(linked_techweb))
 		say("Warning: No linked research system!")
-		return 
-	var/adjusted = orig_light - 10
-	if(adjusted <= 0)
-		say("Explosion not large enough for research calculations.")
 		return
-	var/point_gain = techweb_scale_bomb(adjusted) - techweb_scale_bomb(linked_techweb.max_bomb_value)
-	if(point_gain <= 0)
+
+	var/point_gain = 0
+
+	/*****The Point Calculator*****/
+
+	if(orig_light < 10)
 		say("Explosion not large enough for research calculations.")
 		return
-	linked_techweb.max_bomb_value = adjusted
-	linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, point_gain)
-	say("Gained [point_gain] points from explosion dataset.")
+
+	if(mode == RESEARCH)
+		point_gain = round(((log(orig_light)-1)**1.6)*TECHWEB_BOMB_POINTCAP)
+		linked_techweb.logged_theoretical_points += point_gain
+		say("Logged [point_gain] theoretical points from explosion dataset, new total is [linked_techweb.logged_theoretical_points].")
+
+		if(point_gain > linked_techweb.largest_bomb_value)
+			if(point_gain <= TECHWEB_BOMB_POINTCAP || linked_techweb.largest_bomb_value < TECHWEB_BOMB_POINTCAP)
+				var/old_tech_largest_bomb_value = linked_techweb.largest_bomb_value //held so we can pull old before we do math
+				linked_techweb.largest_bomb_value = point_gain
+				point_gain -= old_tech_largest_bomb_value
+				point_gain = min(point_gain,TECHWEB_BOMB_POINTCAP)
+			else
+				linked_techweb.largest_bomb_value = TECHWEB_BOMB_POINTCAP
+				point_gain = 1000
+
+			linked_techweb.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, point_gain)
+			say("Gained [point_gain] points from explosion dataset.")
+
+		else //you've made smaller bombs
+			say("Data already captured. Aborting.")
+
+	else //Credit production is only capped per bomb
+		point_gain = round(((log(orig_light)-1)**1.6)*TECHWEB_BOMB_POINTCAP)
+		linked_techweb.logged_theoretical_points += point_gain
+		say("Logged [point_gain] theoretical points from explosion dataset, new total is [linked_techweb.logged_theoretical_points].")
+		point_gain = min(point_gain,TECHWEB_BOMB_POINTCAP)
+		SSshuttle.points += point_gain
+		say("Gained [point_gain] credits from explosion dataset.")
+
+	//give the toxins scientists things that directly interest them
+	if(linked_techweb.logged_theoretical_points > 50000)
+		if(linked_techweb.reward_level==1)
+			new /obj/machinery/portable_atmospherics/canister/pluoxium(loc)

```

Comment:
i also disagree with directly giving scientists rewards like pluox/nitryl canisters. i think reward stuff should be gated behind techwebs, like tiered (stock parts) canisters/airpumps that you can research from techweb points directly (which should be handled in a future pr at some point). plus.. directly teleporting in a random canister is a bit.. immersion breaking

also i have never heard of a prototype canister before and i feel like its some unfinished atmoscode so maybe remove that from rewards